basic-example/docker-compose.yml

```diff
@@ -1,19 +1,23 @@
-version: '3'
+version: '3.2'
 services:
+
   zookeeper:
     image: wurstmeister/zookeeper
+    ports:
+      - "2181:2181"
 
   kafka:
-    image: wurstmeister/kafka:1.0.0
-    hostname: kafka
+    image: wurstmeister/kafka:2.11-2.0.0
+    ports:
+      - target: 9092
+        published: 9092
+        protocol: tcp
+        mode: host
     environment:
-      KAFKA_ADVERTISED_HOST_NAME: kafka
+      KAFKA_ADVERTISED_HOST_NAME: $LOCAL_IP
```

Comment:
This didn't work for me, I got errors about the `KAFKA_ADVERTISED_HOST_NAME` and `KAFKA_ADVERTISED_PORT` being deprecated as documented over here https://github.com/wurstmeister/kafka-docker#listener-configuration

`KAFKA_LISTENERS: PLAINTEXT://kafka:9092` seemed to work just fine.

Reply:
If i exchanged LOCAL_IP with the local machines ip it worked. i'll give your suggestion a shot, couldn't get it working yesterday.

Thanks

Reply:
Yea the issue was when contacting kafka from the outside 

`./kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic test-0
[2018-10-09 09:47:17,810] WARN [Consumer clientId=consumer-1, groupId=console-consumer-17355] Error connecting to node kafka:9092 (id: 1001 rack: null) (org.apache.kafka.clients.NetworkClient)
java.io.IOException: Can't resolve address: kafka:9092
	at org.apache.kafka.common.network.Selector.doConnect(Selector.java:235)`

Reply:
Hacky fix - add kafka to your `/etc/hosts` for 127.0.0.1 and everything resolves.... we can leave it aside for this PR. Running kafka in docker always has some caveats.